### DISCUSSION TITLE
Ideas Guidelines

### DISCUSSION THREAD

**Discussion by JohnGuan** [IceWhaleTech/CasaOS#164] [Announcements]
The [Ideas](https://github.com/IceWhaleTech/CasaOS/discussions/categories/ideas) category is mainly for discussing feature requests about CasaOS.

We are very welcome to discuss ideas freely and are eager to hear from people and discuss them together. Also, a good quality idea can make the discussion easier and more efficient for everyone. So we have prepared the following rules. 
If you have better ideas for optimization, feel free to discuss them below!

When opening a new discussion, please ensure that your discussion meets the following requirements.
- Search and browse existing discussions first to avoid duplicate topics.
- To maintain effective community communication, content must be in English
- The content should focus on improvements and new features of CasaOS. Bugs are not included in this category.
- Easy to understand idea titles that start with "Feature:" or "Enhancement:".
- To describe your idea in detail, you can refer to the following template

```
#### Is your idea related to a problem? Please describe:
<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->
- ...

#### Describe the solution you'd like:
<!-- A clear and concise description of what you want to happen.-->
- ...

#### Describe alternatives you've considered:
<!-- A clear and concise description of any alternative solutions or features you've considered.-->
- ...

#### Additional context
<!-- Add any other context or screenshots about the feature request here.-->
- ...
```

[<kbd>💬 **Start a discussion of your idea!**</kdb>](https://github.com/IceWhaleTech/CasaOS/discussions/new?category=ideas)

**Comment by Bobxyo** [comment]
Where is the SSL option？ Can not use https？

**Reply by nghialele** [reply]
Still no answering on this

**Comment by AmadeusGraves** [comment]
The form to insert docker image not manage all variables like -u (User) and others so then why not create a new component that permit insert custom arguments.

![image](https://user-images.githubusercontent.com/18572939/187890745-e68979df-a222-44e8-9b3f-4a6ee7ce74ba.png)

List of arguments, they are so many so its better make a custom argument insert.
https://docs.docker.com/engine/reference/commandline/run/

**Reply by Dethkiller15** [reply]
Some containers need args that aren't accessible in the UI so I second this change

**Reply by worph** [reply]
We really need that especially the "user" option part.

**Comment by WombleWoo7547** [comment]
How about a docker container with CasaOS install pre-installed?? (I do realise that I could do it myself with -v /var/run/docker.sock:/var/run/docker.sock and a ubuntu container and manually install CasaOS on top, however a docker container pre-made will be friendlier)

**Comment by Dvalin21** [comment]
Is there a possibility of adding the use of hardware raid and software raid options?

**Reply by eleazar7** [reply]
Yes! i would like to have this also!

**Reply by MinecraftDocker** [reply]
I would love that!
Instead of having to install it over the top of a more mature NAS operating system like TrueNAS or OMV, you could manage everything from the WebUI. Inclusion off all the major RAID types would be great! And you don't have to worry as much about a future update bricking over-the-top installation and updates.

**Comment by Cp0204** [comment]
建议把可加载第三方应用商店，模式参考群辉；
1、制定appstore规范，如：商店一个json，一个app一个json；
2、用户填入第三方商店源，安装三方应用；

**Comment by zhanghengxin** [comment]
![image](https://user-images.githubusercontent.com/24197448/206615742-690b10c9-d972-4340-97fe-eef513ee92ac.png)
  做成折叠  显示 mac  和流量总计 进出  IPV4   IPV6

**Comment by MISTERNEGATIVE21** [comment]
ADD support for ZIp and Unzip files

**Reply by mousethatischarles** [reply]
CasaOS v0.4.4 Request return of features!
I just upgraded from the previous version to find two features I found very useful now missing:
1) The ability to edit a previously installed Docker or access it's command line via the 'thee dots menu'. (Using Portainer for now)
2) The ability to hide installed dockers not imported to CasaOS - it would also be very helpful to extend this to any 'icon' on the CasaOS desktop. (some installed Dockers have no functionality that may be accessed in this way so just take up space)

As a general comment I'm very much enjoying CasaOS,, so much so I've recently removed OMV from my server.

Suggested improvement / additional features:
1) The addition of wsdd-like functionality to ensure Windows picks up SAMBA shares would be nice. I've installed it and it works.
https://github.com/christgau/wsdd
2) Being able to edit the SMB path and settings when sharing a folder via the File app would be great. For now I've imported my OMV settings directly in to the SAMBA config files, but not ideal for a setup that's supposed to be easy.

Many Thanks,

**Comment by Mattbass123** [comment]
Please can it be made simple to add a MQTT broker in casaos as an app?

**Comment by Felix-Kiefer** [comment]
Hi, could you please add a feature to hide the "Legacy app (To be rebuilt)" section from the main dashboard? It would be way cleaner if just the apps would show up.

**Comment by Aaron-Sauter** [comment]
Hi, is it possible to add some features for customizing the interface, for example folders and different layouts.
That would be very nice.😊

**Comment by borry1973** [comment]
Can you please add an option for a start-up/shutdown timer?
I`ve no need for my nas to run for 24 hours, in openmediavault i can just set a shutdown/startup timer.

**Comment by ljis120301** [comment]
Feature/Enhancement : Add support for those using Private Key based SSH authentication for the web interface. I am unsure of how to access the SSH feature for those using a SSH private and public key rather than password based authentication. This may be a misunderstanding on my part on how to add this but after converting to a key based authentication setup to manage SSH. I am no longer able to connect via the web portal to access ssh remotely. 

How to solve this:

I would recommend giving users the option on the web portal to add their public keys into the web portal such that SSH would connect using a public key rather than with a password

**Comment by thegur90** [comment]
Feature/Enhancement: Add support for subtitles in MKV videos inside the baked in files app. right now i need to download the files to view subtitles)

**Comment by thegur90** [comment]
Feature/Enhancement:
Files app: Compress to ZIP in right-click context menu

**Reply by Xeddius** [reply]
Additionally an advanced mode would be nice. I'd really like to be able to manage permissions for folders.

**Comment by mikepaxton** [comment]
I'd like to be able to hide some of the icons on the Desktop.  I've installed a number of docker apps that don't have or need a webUI, but are actually services.  A couple instances are; Node-Exporter and Cadvisor for displaying server detail in Grafana.  These type of icons just clutter up the desktop and have no use whatsoever.

A simple checkbox in the Icons drop down menu with the title "Hidden" would be great.  Another option might be to use Dockers Label feature and would look something like this: -- label casaos.hidden=true.  Just a simple boolean that could be added to a compose file.

In looking around for this option already being built into CasaOS, i've seen several other people asking for this feature.  I can't imagine someone objecting to this option, as it would really clean up the desktop for people that are running a lot of Docker containers.

**Comment by TomYule** [comment]
Feature Request: CPU & Drive Temperature Monitoring
First of all, thank you for the great work on CasaOS! It's been a wonderful experience so far.

I’d like to suggest a future enhancement: support for displaying CPU temperature and hard drive/SSD temperatures in the system monitoring section.

This would be very helpful for users who want to keep an eye on their system health, especially in small servers or home setups where cooling may be limited.

For example:

    CPU temperature (overall and per-core if possible)

    Drive temperatures for connected HDDs/SSDs/NVMe drives

This could be shown in the existing resource widget or in a dedicated “Hardware Health” panel.

Thanks for considering this suggestion!


armbian

```
    _             _    _              ___  ___ 
   /_\  _ _ _ __ | |__(_)__ _ _ _    / _ \/ __|
  / _ \| '_| '  \| '_ \ / _` | ' \  | (_) \__ \
 /_/ \_\_| |_|_|_|_.__/_\__,_|_||_|  \___/|___/
                                               
 v25.11.0 for Aml.S922x running Armbian Linux 6.12.59-ophub

 Packages:     Debian stable (bookworm)
 Updates:      Kernel upgrade enabled and 2 packages available for upgrade 
 IPv4:         (LAN) 192.168.31.11 
 IPv6:         fd00:6868:6868::342, fd00:6868:6868:0:b66c:a947:18fa:79be 
 Containers:   immich_server,immich_postgres immich_machine_learning,immich_redis cloudflared

 Performance:  

 Load:         170%              Uptime:       18 hours, 21 minutes
 Memory usage: 44% of 3.58G      Zram usage:    12% of 1.79G  
 CPU temp:     59°C              Usage of /:   26% of 6.7G   
 storage /:    66% of 916G   
 RX today:     18 MiB           

 Commands: 

0m           

 Commands: 

 Configuration : armbian-config
 Upgrade       : armbian-upgrade
 Monitoring    : htop

root@armbian:~# cat /sys/devices/virtual/thermal/thermal_zone0/temp
60100
root@armbian:~# 


10T09:27:00.041+0800	info	CPU thermal zone found: cpu-thermal, path: /sys/devices/virtual/thermal/thermal_zone0.	{"func": "service.GetCPUThermalZone", "file": "/home/runner/work/CasaOS/CasaOS/service/system.go", "line": 485}
10T09:32:00.043+0800	info	CPU thermal zone found: cpu-thermal, path: /sys/devices/virtual/thermal/thermal_zone0.	{"func": "service.GetCPUThermalZone", "file": "/home/runner/work/CasaOS/CasaOS/service/system.go", "line": 485}
10T09:37:05.043+0800	info	CPU thermal zone found: cpu-thermal, path: /sys/devices/virtual/thermal/thermal_zone0.	{"func": "service.GetCPUThermalZone", "file": "/home/runner/work/CasaOS/CasaOS/service/system.go", "line": 485}

```

**Comment by TomYule** [comment]
[Feature Request] Add Manual Network Configuration Support for IPv4/IPv6 including IP, DNS, Gateway, Proxy and Protocol Preference Selection
I would like to make a suggestion: I hope CasaOS can support both IPv4 and IPv6 with manual configuration options for IP address, DNS, gateway, and proxy settings, as well as provide a network preference selection feature to choose which protocol to prioritize when both IPv4 and IPv6 are available.

**Comment by messengerstinxx** [comment]
**[Installation Guide] Latest Docker Release Fix » CasaOS Install » Custom App Store**
_I'm uncertain if I'm posting in the correct area, however this should resolve the issue(s) many face when performing an "apt update"._

=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
▒▒▒▒▒ INSTALL DOCKER ENGINE ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒
➭ REF. VIDEO: https://www.youtube.com/watch?v=mN8DC54UtM8
➭ REF. DOCS: https://docs.docker.com/engine/install/debian/

➭ SSH into the device and elevate to ROOT to complete

➭ Run the following command to uninstall all conflicting packages:
apt remove $(dpkg --get-selections docker.io docker-compose docker-doc podman-docker containerd runc | cut -f1)

➭ Set up and install Docker Engine from
➭ REF: https://docs.docker.com/engine/install/debian/#install-using-the-repository
apt update -y

apt install -y ca-certificates curl

install -m 0755 -d /etc/apt/keyrings

curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc

chmod a+r /etc/apt/keyrings/docker.asc

➭ Add the repository to Apt sources (copy/paste this entire statement):

➭ ┌─── COPY & PASTE CONTENT WITHIN THIS FIELD AS A SINGLE COMMAND ───┐
sudo tee /etc/apt/sources.list.d/docker.sources <<EOF
Types: deb
URIs: https://download.docker.com/linux/debian
Suites: $(. /etc/os-release && echo "$VERSION_CODENAME")
Components: stable
Signed-By: /etc/apt/keyrings/docker.asc
EOF
➭ └─── COPY & PASTE CONTENT WITHIN THIS FIELD AS A SINGLE COMMAND ───┘

➭ UPDATE the package list again
apt update -y

➭ To install the latest version, run:
apt install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin

➭ The Docker service starts automatically after installation. To verify that Docker is running, use:
systemctl status docker

➭ [CTRL + C] to end...
➭ Confirm the version installed
docker --version

➭ Verify that the installation is successful by running the hello-world image:
docker run hello-world


▒▒▒▒▒ INSTALL CASA-OS ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒

➭ REF. VIDEO: https://www.youtube.com/watch?v=FwJByjTdKks&ab_channel=DBTech

➭ Estimated installation time (3 to 5 minutes) 
➭ Enter the following command into a clean headless install of Debian, Raspbian, Ubuntu, etc.
cd /opt

curl -fsSL https://get.casaos.io | sudo bash

=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

▒▒▒▒▒ ADD CASA-OS APP STORE REPOSITORIES ▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒

➭ REF. URL: https://awesome.casaos.io/content/3rd-party-app-stores/list.html#_1-casaos-linuxserver-appstore

➭ Estimated installation time (3 to 5 seconds) 
➭ LOGIN to CasaOS
➭ CLICK on App Store within Casa OS
➭ CLICK on the dropdown menu next to the number of available apps
➭ CLICK on More Apps
➭ ENTER either of the.zip links listed below
➭ CILCK [+ Add]

➭ BigBear Repository
https://github.com/bigbeartechworld/big-bear-casaos/archive/refs/heads/master.zip

➭ CasaOS-Coolstore
https://casaos-appstore.paodayag.dev/coolstore.zip

➭ LinuxServer.IO Repository
https://casaos-appstore.paodayag.dev/linuxserver.zip

=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

... Hope this helps.  I use CasaOS on various architectures including RaspberryPi and ASUS Tinkerboards.  When I discovered that the project jumped ship to focus on ZimaOS I almost considered that platform as well, until I saw its Build Requirement Limitations.

CasaOS IMO is an excellent docker container hosting platform for Raspberry Pi's as I even did a combo build of a 8gb Pi hosting a SAMBA DC (i.e.:DNS2) running CasaOS and Webmin.  Then used Portainer to create a MACVLAN that runs Technitium AD-Blocking (i.e.:DNS1) that replaced my Windows AD DC DNS host for public/private DNS hosting.  This could've been accomplished on a Pi4 4gb, however the 8gb also hosts a total of 10 other CasaOS Docker containers with resources to spare.

![RPI CasaOS](https://github.com/user-attachments/assets/2361b0b2-6a01-446b-9f09-bebfeaee98b2)

![RPI SAMBA DC](https://github.com/user-attachments/assets/f31f03b6-7206-4e15-9f93-d640ad864a23)
